apis_bibsonomy/models.py

```diff
@@ -29,3 +32,25 @@ def bibtexjson(self):
 
     def get_absolute_url(self):
         return reverse("apis_bibsonomy:referencedetail", kwargs={"pk": self.pk})
+
+    def save(self):
+        bib_ref = self.bibs_url
+        sources = getattr(settings, "APIS_BIBSONOMY", [])
+        sources = [s for s in sources if s.get("url") and s.get("url") in self.bibs_url]
+
+        source = sources[-1]
+
+        if self.bibs_url is not None:
+            btype = source.get("type", None)
+            if btype == "bibsonomy":
+                bib_e = BibsonomyEntry(bib_hash=bib_ref, base_set=sett1)
```

Comment:
sett1 is not defined here, see this snippet for comparison:

[```python 
   sett = getattr(settings, "APIS_BIBSONOMY", [])
        for s in sett:
            if "url" in s.keys():
                if s["url"] in bib_ref:
                    sett1 = s
                    kind = s["type"]
        if bib_ref is not None:
            r = {"bibs_url": bib_ref}
            if kind == "bibsonomy":
                bib_e = BibsonomyEntry(bib_hash=bib_ref, base_set=sett1)
```](https://github.com/acdh-oeaw/apis-bibsonomy/blob/878eff3382dd4da74aed33db8417fc872582cffc/apis_bibsonomy/api_views.py#L31C3-L52)